nncf/common/tensor_statistics/aggregator.py

```diff
@@ -82,6 +84,11 @@ def collect_statistics(self, model: TModel, graph: NNCFGraph) -> None:
             empty_statistics = False
         if empty_statistics:
             raise nncf.ValidationError(EMPTY_DATASET_ERROR)
+        
+        if len(self.dataset) < subset_size:
```

Comment:
There is no guarantee that `__len__` is implemented for the dataset. In the `Statistics collection` loop, you can count the number of processed input_data, and after the loop, you can check if `subset_size` is higher than the number of samples in the provided dataset. I think we should display a warning instead of an error in this case.